### PR TITLE
refactor: clean up bit

### PR DESCRIPTION
**Pull request by bwanglzu**
1. more unittest with edge cases
3. remove `default` from constructor
4. readme polish

### CODE REVIEW

**Comment by bwanglzu** [comment]
fixing the test

File: jinahub/encoders/image/BigTransferEncoder/README.md [hunk 1]
```diff
@@ -1,38 +1,57 @@
...
 pods:
   - name: encoder
-    uses: 'jinahub+docker://BigTransferEncoder/gpu'
-    gpus: all
-    uses_with:
-      device: '/GPU:0'
+    uses: 'jinahub+docker://BigTransferEncoder'
 ```
 
+Note that this way the Executor will download the model every time it starts up. You can
```
Comment (cristianmtr) [inline jinahub/encoders/image/BigTransferEncoder/README.md:26]: Just link to the guide in docs

**Review by nan-wang — APPROVED** [review]
LGTM👍
